lib/node/nodes/weighted-centroid.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+var Node = require('../node');
+var dot = require('dot');
+
+var TYPE = 'weighted-centroid';
+
+var PARAMS = {
+    source : Node.PARAM.NODE(Node.GEOMETRY.POINT),
+    weight_column : Node.PARAM.STRING,
+    category_column : Node.PARAM.STRING
+};
+
+var WeightedCentroid  = Node.create(TYPE, PARAMS, {cache: false});
+
+module.exports = WeightedCentroid;
+module.exports.TYPE = TYPE;
+module.exports.PARAMS = PARAMS;
+
+var weightedCentroidTemplate = dot.template([
+    'SELECT the_geom , class',
```

Comment:
Oh, thanks. Understood.

What about producing a cartodb_id column? Other analyses might need that. For instance, and I know it's a silly example, kmeans require to have a cartodb_id column: if I apply a `kmeans` over the result of `weighted-centroid` it will fail because `weighted-centroid` is not producing a cartodb_id column.